packages/cardano-services/src/Asset/TypeormAssetProvider/TypeormAssetProvider.ts

```diff
@@ -0,0 +1,44 @@
+import { Asset, AssetProvider, Cardano, GetAssetArgs } from '@cardano-sdk/core';
+import { NftMetadataService } from '../types';
+import { TypeormProvider, TypeormProviderDependencies } from '../../util';
+
+interface TypeormAssetProviderDependencies extends TypeormProviderDependencies {
+  nftMetadataService: NftMetadataService;
+}
+
+export class TypeormAssetProvider extends TypeormProvider implements AssetProvider {
+  #dependencies: TypeormAssetProviderDependencies;
+
+  constructor(dependencies: TypeormAssetProviderDependencies) {
+    const { connectionConfig$, entities, logger } = dependencies;
+    super('TypeormAssetProvider', { connectionConfig$, entities, logger });
+
+    this.#dependencies = dependencies;
+  }
+
+  async getAsset({ assetId }: GetAssetArgs): Promise<Asset.AssetInfo> {
+    const assetName = Cardano.AssetId.getAssetName(assetId);
+    const policyId = Cardano.AssetId.getPolicyId(assetId);
+    const nftMetadata = await this.#dependencies.nftMetadataService.getNftMetadata({
```

Comment:
Same for `tokenMetadata`. To resolve it, add `TokenMetadataService` to `TypeormAssetProviderDependencies`
```suggestion
  async getAsset({ assetId, nftMetadata: needNftMetadata }: GetAssetArgs): Promise<Asset.AssetInfo> {
    const assetName = Cardano.AssetId.getAssetName(assetId);
    const policyId = Cardano.AssetId.getPolicyId(assetId);
    const nftMetadata = needNftMetadata ? await this.#dependencies.nftMetadataService.getNftMetadata(...) : null
```

Reply:
Fixed on 5f0b649